### PR TITLE
Deneb refactoring and new tests

### PR DESCRIPTION
**Pull request by hwwhww**
- Minor spec changes:
    - Add explicit `assert (len(opaque_tx) - blob_versioned_hashes_offset) % 32 == 0` check in `tx_peek_blob_versioned_hashes` for better error detection
        - Otherwise, if we send a tx with incorrect length, the error will be a Remerkleable deserialization error. 
    - Modify `process_blob_kzg_commitments` function signature by removing `state`
        - It seems no obvious reason to keep it.
- Add new `deneb/sanity/test_blocks.py` test cases.

### CODE REVIEW

File: tests/core/pyspec/eth2spec/test/deneb/sanity/test_blocks.py [hunk 1]
```diff
@@ -16,32 +18,134 @@
...
 
 
+def run_block_with_blobs(spec, state, blob_count):
+    yield 'pre', state
+
+    block = build_empty_block_for_next_slot(spec, state)
+    opaque_tx, _, blob_kzg_commitments, _ = get_sample_opaque_tx(spec, blob_count=blob_count)
+    block.body.blob_kzg_commitments = blob_kzg_commitments
+    block.body.execution_payload.transactions = [opaque_tx]
+    block.body.execution_payload.excess_data_gas = 2
```
Comment (asn-d6) [inline tests/core/pyspec/eth2spec/test/deneb/sanity/test_blocks.py:28]: Just to understand, here we are testing that any non-zero `excess_data_gas` value will be accepted?

Comment (hwwhww) [reply tests/core/pyspec/eth2spec/test/deneb/sanity/test_blocks.py:28]: @asn-d6 correct.

**Review by djrtwo — COMMENTED** [review]
nice!

File: specs/deneb/beacon-chain.md [hunk 2]
```diff
@@ -173,6 +173,8 @@ def tx_peek_blob_versioned_hashes(opaque_tx: Transaction) -> Sequence[VersionedH
         message_offset
         + uint32.decode_bytes(opaque_tx[(message_offset + 188):(message_offset + 192)])
     )
+    # `VersionedHash` is a 32-byte object
+    assert (len(opaque_tx) - blob_versioned_hashes_offset) % 32 == 0
```
[the remark this answers is not in the document]
Comment (hwwhww) [reply specs/deneb/beacon-chain.md:177]: It's the total length of `blob_versioned_hashes: List[VersionedHash, MAX_VERSIONED_HASHES_LIST_SIZE]` so it is multiples of 32.

**Review by djrtwo — APPROVED** [review]

**Pull request merged**